hugegraph-dist/pom.xml

```diff
@@ -179,21 +179,42 @@
                             <goal>run</goal>
                         </goals>
                         <configuration>
-                            <tasks>
-                                <echo file="${project.basedir}/dist.sh">
-                                    wget --version 1>/dev/null || exit
-                                    wget https://github.com/swagger-api/swagger-ui/archive/refs/tags/v4.15.5.tar.gz
-                                    tar zxvf v4.15.5.tar.gz
-                                    echo "window.onload = function() { window.ui = SwaggerUIBundle({
-                                    url:'/openapi.json',dom_id:'#swagger-ui',deepLinking:true,layout:'StandaloneLayout',
-                                    presets:[SwaggerUIBundle.presets.apis, SwaggerUIStandalonePreset ],
-                                    plugins:[SwaggerUIBundle.plugins.DownloadUrl]});};" > swagger-ui-4.15.5/dist/swagger-initializer.js
-                                    cp -r swagger-ui-4.15.5/dist ../${final.name}/swagger-ui
-                                </echo>
-                                <exec executable="${shell-executable}" dir="${project.basedir}" failonerror="true">
-                                    <arg line="./dist.sh"/>
-                                </exec>
-                            </tasks>
+                            <target>
+                                <taskdef resource="net/sf/antcontrib/antcontrib.properties"
+                                         classpathref="maven.plugin.classpath"/>
+                                <if>
```

Comment:
maybe we could also check if curl exist in os env？ skip the swagger building if network/curl broken（do we need it？）

In general，the building shouldn't be bounded with outside network？

Reply:
@imbajin `curl` has been checked in [`3ae4aac` (#2277)](https://github.com/apache/incubator-hugegraph/pull/2277/commits/3ae4aacd33bcd9ee7d3913fd88683201ea4272f1#diff-5cb24cae8e78777b5af60640cc24189b37905e8d9229fcdc25298ce185caca88R190)

If there is no outside network, how can maven download dependencies from remote repositories during the building process?

Reply:
- I mean check but not exit if miss it（skip instead）
- like many machine could use inner “self-host repository”（like gitlab VS github） to download deps，and next time they will reuse it from local .m2 repo

Reply:
@imbajin Using this snippet without having curl installed:

```bash
curl --version >/dev/null 2>&1 || (echo 'WARN: Please install `curl` first' && exit)
```

Does the following result meet your expectations?

![image](https://github.com/apache/incubator-hugegraph/assets/79143929/64245752-a220-4ab1-b2d1-50472ca9ea60)

Similar outcomes can be expected in the case of network abnormalities.

Reply:
> Does the following result meet your expectations?
> 
> ![image](https://user-images.githubusercontent.com/79143929/260348165-64245752-a220-4ab1-b2d1-50472ca9ea60.png)
> 
> Similar outcomes can be expected in the case of network abnormalities.

LGTM, the warb message could  be "ERROR: Please install `curl` first if you need `swagger-ui`"

Reply:
> > Does the following result meet your expectations?
> > ![image](https://user-images.githubusercontent.com/79143929/260348165-64245752-a220-4ab1-b2d1-50472ca9ea60.png)
> > Similar outcomes can be expected in the case of network abnormalities.
> 
> LGTM, the warning message could be "ERROR: Please install `curl` first if you need `swagger-ui`"
> 
> BTW, could also skip download if network bad like `curl xxx || exit`?)

Sure, the corresponding modifications can be found in [`0a9215f` (#2277)](https://github.com/apache/incubator-hugegraph/pull/2277/commits/0a9215f5ccc83dcacaf80142c0910a58039323be)